src/main/java/wolox/training/exceptions/handler/ErrorHandler.java

```diff
@@ -0,0 +1,33 @@
+package wolox.training.exceptions.handler;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.exceptions.BookNotFoundException;
+
+@ControllerAdvice
+public class ErrorHandler {
+
+    @ExceptionHandler(BookNotFoundException.class)
+    @ResponseStatus(value = HttpStatus.NOT_FOUND)
+    @ResponseBody
+    protected ErrorResponse BookNotFoundException(BookNotFoundException ex) {
+
+        return new ErrorResponse(ErrorsEnum.JSON_BOOK_NOT_FOUND_EXCEPTION.getCode(),
+                ErrorsEnum.JSON_BOOK_NOT_FOUND_EXCEPTION.getMessage());
+    }
+
+    @ExceptionHandler(BookIdMismatchException.class)
+    @ResponseStatus(value = HttpStatus.NOT_FOUND)
```

Comment:
I would say this is a BAD_REQUEST rather than NOT_FOUND. What do you think? 🙂

Reply:
Yes you're right. Thanks!!